### PR TITLE
Fix public viewing link

### PR DESCRIPTION
**Pull request by A-ryan-Kalra**
I fixed the code by adding a timer for one second. Without this, setLoading would be set to false upon component re-render, causing the page to display error 404 by default. Therefore, it needed time to wait until hydration is synced and process the verification accordingly

### CODE REVIEW

File: pages/[username]/posts/[postId].tsx [hunk 1]
```diff
@@ -64,6 +63,7 @@ export const PostPage = ({}: Props) => {
       const noteDoc = doc(db, "users", user.uid, "notes", postId as string);
       const noteSnapshot = await getDoc(noteDoc);
       note = noteSnapshot.data() as NoteDocument;
+      console.log(note, " note");
```
Comment (NayamAmarshe) [inline pages/[username]/posts/[postId].tsx:66]: No console logs in PRs please.

Comment (NayamAmarshe) [inline pages/[username]/posts/[postId].tsx:66]: No console logs in PRs please.

File: pages/[username]/posts/[postId].tsx [hunk 2]
```diff
@@ -89,8 +89,11 @@ export const PostPage = ({}: Props) => {
   };
 
   useEffect(() => {
-    fetchData();
-  }, [router.query]);
+    const timer = setTimeout(() => {
+      fetchData();
+    }, 1000);
+    return () => clearTimeout(timer);
+  }, [username, postId]);
```
Comment (NayamAmarshe) [inline pages/[username]/posts/[postId].tsx:96]: We do not want to be using a timeout inside useEffect. A timeout does not scale according to the internet speed. Would you have another way to solve the issue?

File: pages/[username]/posts/[postId].tsx [hunk 3]
```diff
@@ -26,10 +26,9 @@ export const PostPage = ({}: Props) => {
   const [name, setName] = useState<string>("");
   const [profilePicture, setProfilePicture] = useState<string>("");
   const [loading, setLoading] = useState(true);
+  const { username, postId } = router.query;
```
Comment (NayamAmarshe) [inline pages/[username]/posts/[postId].tsx:29]: Does moving the `router.query` outside of `fetchData()` change the behavior in any way?

**Review by NayamAmarshe — CHANGES_REQUESTED** [review]
Thanks for the changes. Could you please take a look at these suggestions?

File: pages/[username]/posts/[postId].tsx (hunk 2 above)
Comment (A-ryan-Kalra) [reply pages/[username]/posts/[postId].tsx:96]: No worries this time I have implemented conditionals that will execute functions based on specific conditions.

**Comment by NayamAmarshe** [comment]
LGTM!

**Review by NayamAmarshe — APPROVED** [review]

**Pull request merged**